src/edu/csus/ecs/pc2/ui/ClarificationsTablePane.java

```diff
@@ -1114,7 +1130,29 @@ protected void requestSelectedClarification() {
         try {
             ElementId elementId = clarificationTable.getElementIdFromTableRow(selectedIndexes[0]);
             Clarification clarificationToAnswer = getContest().getClarification(elementId);
-
+            if (elementId.equals(lastAnswered)) {
+                // try to make it visible
+                answerClarificationFrame.setVisible(true);
+                return;
+            }
+            if (lastAnswered != null) {
+                int showConfirmDialog = JOptionPane.showConfirmDialog(getParentFrame(), "You are asking to answer a new clarification, but you already have a clarification checked out for answering.\n" + //
+                        "\n" + //
+                        "If you want to give up the previous clarification and switch to answering the new one, click \"Yes\";\n" + //
+                        "\n" + //
+                        "otherwise, click \"No\" and go back to answering the previous clarification.\n" + //
+                        "", "Already answering a Clarifition", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE, null);
+                if (showConfirmDialog == JOptionPane.NO_OPTION) {
```

Comment:
The above line causes a bug: if the user hits the "Closing X" on the dialog frame, the value returned from the `showConfirmDialog()` call is `-1` (undefined).  Since `-1 != JOptionPane.NO_OPTION`, hitting the "Close X" is the same thing as saying "Yes (I want to switch)" -- and so it switches to the new clar (which would, I think, surprise the user).  The test here should be `showConfirmDialog != JOptionPane.YES_OPTION`.

To reproduce:  go through the "Steps to test the PR", including checking out one clar for answering and then attempting to check out a second clar for answering.  When the following dialog appears:

![image](https://github.com/pc2ccs/pc2v9/assets/9256691/e8540cf9-1d31-4f12-b6cb-1114648f7c75)

hit the "X" in the upper right corner.

Expected result:   no change in the state of the system.
Actual result:  judging will switch to the new clar.

Also: I think the message in the title of the dialog should say "Clarification" instead of "Clarifition"...

Reply:
good catches.

Reply:
fixed both items.